setup.py

```diff
@@ -1008,18 +1055,32 @@ def debug_build() -> bool:
 
 
 files: list[str | os.PathLike[str]] = ["src/_imaging.c"]
+define_macros: list[tuple[str, str | None]] = []
+
+tls_define_macro = _try_compile_tls_define_macro(distutils.ccompiler.new_compiler())
+if tls_define_macro is not None:
+    define_macros.append(tls_define_macro)
+
 for src_file in _IMAGING:
     files.append("src/" + src_file + ".c")
 for src_file in _LIB_IMAGING:
     files.append(os.path.join("src/libImaging", src_file + ".c"))
 ext_modules = [
-    Extension("PIL._imaging", files),
```

Comment:
Turns out that inside `build_extensions` there's an `self._update_extension` call that updates the defines, libraries, sources etc. We should do that instead of calling `_try_compile_tls_define_macro` here again and then passing those in the `Extension` call.

The only problem is with `_imagingmath` and `_imagingmorph` for which there's no calls to `_update_extension`. But we could add those in with very minimal changes to the function signature (to allow for calling it without `libs`).

Reply:
Got it! Just updated it.